server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMailboxesMethodTest.java

```diff
@@ -2087,29 +2090,98 @@ public void setMailboxesShouldReturnNotUpdatedErrorWhenMovingMailboxTriggersName
+    }
+
+    @Test
+    public void delegateMailboxShouldUpdateParentMailboxAsLookupRight() throws Exception {
+        String shared = "shared";
+        String sharedSubMailbox = "shared.subMailbox";
+        String user = "sharee";
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, username, shared);
+        MailboxId mailboxId = mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, username, sharedSubMailbox);
+
+        with()
+            .header("Authorization", accessToken.serialize())
+            .body("[" +
+                "  [ \"setMailboxes\"," +
+                "    {" +
+                "      \"update\": {" +
+                "        \"" + mailboxId.serialize() + "\" : {" +
+                "          \"sharedWith\" : {\"" + user + "\": [\"a\", \"w\"]}" +
+                "        }" +
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]")
+            .post("/jmap");
+
+        awaitMethod.run();
+
+        MailboxACL mailboxACL = aclProbe.retrieveRights(MailboxPath.forUser(username, shared));
+
+        assertThat(mailboxACL.getEntries())
+            .contains(new MailboxACL.Entry(user, "l"));
+    }
+
+    @Test
+    public void delegateMailboxShouldUpdateAllParentMailboxesAsLookupRight() throws Exception {
+        String shared = "shared";
+        String sharedSubMailbox1 = "shared.subMailbox1";
+        String sharedSubMailbox2 = "shared.subMailbox1.subMailbox2";
+        String user = "sharee";
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, username, shared);
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, username, sharedSubMailbox1);
+        MailboxId mailboxId = mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, username, sharedSubMailbox2);
+
+        with()
+            .header("Authorization", accessToken.serialize())
+            .body("[" +
+                "  [ \"setMailboxes\"," +
+                "    {" +
+                "      \"update\": {" +
+                "        \"" + mailboxId.serialize() + "\" : {" +
+                "          \"sharedWith\" : {\"" + user + "\": [\"a\", \"w\"]}" +
+                "        }" +
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]")
+            .post("/jmap");
+
+        awaitMethod.run();
+
+        MailboxACL mailboxACL1 = aclProbe.retrieveRights(MailboxPath.forUser(username, shared));
+        MailboxACL mailboxACL2 = aclProbe.retrieveRights(MailboxPath.forUser(username, sharedSubMailbox1));
+
+        assertThat(mailboxACL1.getEntries())
+            .contains(new MailboxACL.Entry(user, "l"));
+        assertThat(mailboxACL2.getEntries())
+            .contains(new MailboxACL.Entry(user, "l"));
+    }
+}
```

Comment:
What if I delegate to 2 users? Would they both be updated?

Reply:
Yes, added cucumber testing